apollo-router/src/plugins/telemetry/config_new/connector/events.rs

```diff
@@ -46,92 +58,67 @@ pub(crate) fn new_connector_events(
     let custom_events = config
         .custom
         .iter()
-        .filter_map(|(event_name, event_cfg)| match &event_cfg.level {
-            EventLevel::Off => None,
-            _ => Some(CustomEvent {
-                inner: Mutex::new(CustomEventInner {
-                    name: event_name.clone(),
-                    level: event_cfg.level,
-                    event_on: event_cfg.on,
-                    message: event_cfg.message.clone(),
-                    selectors: event_cfg.attributes.clone().into(),
-                    condition: event_cfg.condition.clone(),
-                    attributes: Vec::new(),
-                    _phantom: Default::default(),
-                }),
-            }),
-        })
+        .filter_map(|(name, config)| CustomEvent::from_config(name, config))
         .collect();
 
     ConnectorEvents {
-        request: config.attributes.request.clone().into(),
-        response: config.attributes.response.clone().into(),
-        error: config.attributes.error.clone().into(),
+        request: StandardEvent::from_config(&config.attributes.request),
+        response: StandardEvent::from_config(&config.attributes.response),
+        error: StandardEvent::from_config(&config.attributes.error),
         custom: custom_events,
     }
 }
 
-impl Instrumented
-    for CustomEvents<
-        ConnectorRequest,
-        ConnectorResponse,
-        (),
-        ConnectorAttributes,
-        ConnectorSelector,
-    >
-{
-    type Request = ConnectorRequest;
-    type Response = ConnectorResponse;
-    type EventResponse = ();
-
-    fn on_request(&self, request: &Self::Request) {
+impl CustomEvents<ConnectorRequest, ConnectorResponse, (), ConnectorAttributes, ConnectorSelector> {
```

Comment:
I removed the `Instrumented` trait from here, and it worked without any other code modifications.
Doing this allowed me to change `on_request` so it now gets a mutable reference.